packages/core/apps/core-apps-server-internal/assets/legacy_theme.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/* eslint-disable no-var */
+
+function systemIsDark() {
+  try {
+    return window.matchMedia('(prefers-color-scheme: dark)').matches;
+  } catch (e) {
+    return false;
+  }
+}
+
+function createInlineStyles(content) {
+  var head = document.getElementsByTagName('head')[0];
+  var style = document.createElement('style');
+  style.textContent = content;
+  head.appendChild(style);
+}
+
+// must be kept in sync with
+// packages/core/rendering/core-rendering-server-internal/src/views/styles.tsx
+
+var lightStyles = [
+  'html { background-color: #F8FAFD; }',
+  '.kbnWelcomeText { color: #69707D; }',
+  '.kbnProgress { background-color: #F5F7FA; }',
+  '.kbnProgress:before { background-color: #006DE4; }',
+].join('\n');
+
+var darkStyles = [
+  'html { background-color: #141519; }',
+  '.kbnWelcomeText { color: #98A2B3; }',
+  '.kbnProgress { background-color: #25262E; }',
+  '.kbnProgress:before { background-color: #1BA9F5; }',
+].join('\n');
+
+if (systemIsDark()) {
+  createInlineStyles(darkStyles);
+} else {
+  createInlineStyles(lightStyles);
+}
```

Comment:
So this was required to work around point `4. Static inline styles` from the issue's description.

- we need those theme-dependent styles as soon as the loading page
- Core isn't fully up yet, and doing that logic in the bootstrap script is also too late
- Because of the CSP config, I can't inline a script in the template
- Meaning that I can't know, from this script, what the value of the `darkMode` setting is.

So the only solution I found was:

- if darkMode is `true` or `false`
  - then keep the inline styles in the template
  - do not inject that script
- if darkMode is `system`
  - do not inject the line styles in the template
  - inject that script
  
I would have loved to have a single place to do that, but until we properly implement an `nonce-` mechanism (https://github.com/elastic/kibana/issues/93785) to allow executing inline script from the page template, this is the only viable approach